File: Javascript/Algorithms/Sorting/quickSort.js
function change(items, leftIndex, rightIndex){
    var temp = items[leftIndex];
    items[leftIndex] = items[rightIndex];
    items[rightIndex] = temp;
}
function division(items, left, right) {
    var pivot   = items[Math.floor((right + left) / 2)], //middle element
        i       = left, 
        j       = right;
    while (i <= j) {
        while (items[i] < pivot) {
            i++;
        }
        while (items[j] > pivot) {
            j--;
        }
        if (i <= j) {
            change(items, i, j);
            i++;
            j--;
        }
    }
    return i;
}

function quickSort(items, left, right) {
    var index;
    if (items.length > 1) {
        index = division(items, left, right); 
        if (left < index - 1) {
            quickSort(items, left, index - 1);
        }
        if (index < right) {
            quickSort(items, index, right);
        }
    }
    return items;
}

/**
 * Example:
 */

const array = [0,19,22,75,13,16,12,123,55,2,1,64];

const sortedArray = quickSort(array, 0, array.length - 1);
console.log(sortedArray);

File: Julia/Algorithms/Sorting/quicksort.jl
function quicksort(array, start=1, stop=length(array))

    left, right = start, stop
    if (right <= left)
        return
    end
    pivot = array[convert(Int,floor((left + right) / 2))]
    
    while left <= right
        while array[left] < pivot
            left += 1
        end
        while array[right] > pivot
            right -= 1
        end
        if left <= right
            array[left], array[right] = array[right], array[left]
            left += 1
            right -= 1
        end
    end
    quicksort(array, start, right)
    quicksort(array, left, stop)
    return array
end
arr = [rand(-1000:1000) for i in 1:200]
println(arr)
quicksort(arr)
println(arr)